### ISSUE TITLE
Search.php Failure on php 8.3

### ISSUE THREAD

**Issue by MWE001** [IntegraMOD/phpBB2x#12]
Search.php crashed with epic fashion and style!

I got on it and changed a block of code. Was it right? Meh. At this time, I will say it is 6 in one and a half dozen in the other. It fixed the search and made it function so that is good, but now it if falls right back into the mysqli error that I posted an issue about earlier today. [https://github.com/IntegraMOD/phpBB2x/issues/10](url)

The block of code throwing the error was: 

`for($i = 0; $i < count($store_vars); $i++)
        {
            if (isset(${$store_vars}[$i]))
            {
                $store_search_data[$store_vars[$i]] = ${$store_vars}[$i];
            }
			else
			{
                $store_search_data[$store_vars[$i]] = '';
            }
        }`
        
        I changed it to the following and now the search works but now we get the mysqli error (acts like the db is disconnecting).
        
        `for($i = 0; $i < count($store_vars); $i++)
		{
			$store_search_data[$store_vars[$i]] = ( isset(${$store_vars[$i]}) ? ${$store_vars[$i]} : '' );
		}`

**Comment by MWE001** [comment]
That's about all I have in me today. I have a few things I fixed I am going to push. Have a good night fellas.

**Comment by MWE001** [comment]
This whole db closing thing is apparently above my pay grade boys, I thought just maybe it may have something to do with tight Stack settings so I adjusted the timeout threshold and MySQL Memory Limits and I ended up with a whole lotta notta. Not a dang thing. I will take a screenshot to verify with you guys what I am seeing just in case you try to replicate it. 

Oh and on a side note, ........ I need to have a private chat about this matter due to a matter that is better off behind closed curtains. I'll take it over to the main website.
![Screenshot_5](https://github.com/user-attachments/assets/b4c28719-f321-485c-be85-c74ce5df8d55)
